File: app/src/main/java/com/group15/fitnesstracker/db/Trains.java
package com.group15.fitnesstracker.db;

import androidx.room.Entity;
import androidx.room.ForeignKey;

@Entity(foreignKeys = {@ForeignKey(entity = Trainer.class,parentColumns = "trainerId",childColumns = "trainerId"),
@ForeignKey(entity = User.class,parentColumns = "id",childColumns = "userId")})
public class Trains {
    public int trainerId;
    public int userId;
}

File: app/src/main/java/com/group15/fitnesstracker/db/Trains.kt
package com.group15.fitnesstracker.db

import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(primaryKeys = ["trainerid, userid"],
        foreignKeys = [
            ForeignKey(entity = Trainer::class, parentColumns = arrayOf("trainerId"), childColumns = arrayOf("trainerId")),
            ForeignKey(entity = User::class, parentColumns = arrayOf("id"), childColumns = arrayOf("userId"))
        ])
data class Trains (
    var trainerId: Int,
    var userId: Int
)
